WooCommerce/WooCommerceTests/POS/ViewModels/TotalsViewModelTests.swift

```diff
@@ -22,8 +24,37 @@ final class TotalsViewModelTests: XCTestCase {
                               currencyFormatter: .init(currencySettings: .init()),
                               paymentState: .acceptingCard,
                               isSyncingOrder: false)
+        cancellables = []
+    }
+    override func tearDown() {
+        cancellables.forEach {
+            $0.cancel()
+        }
+        cancellables.removeAll()
+
+        super.tearDown()
+    }
+    func test_isSyncingOrder() async throws {
+        // Given
+        let item = Self.makeItem()
+        let expectation = XCTestExpectation(description: "Expect isSyncingOrder to be true after syncOrder is called")
+        var expectedSyncing = false
+        XCTAssertEqual(sut.isSyncingOrder, expectedSyncing)
+
+        sut.$isSyncingOrder
+            .dropFirst()
+            .sink { value in
+                XCTAssertEqual(value, expectedSyncing)
```

Comment:
I see that CI passes but for some reason this test is asserting twice and failing in my local environment. Can you attempt to run it several times locally in a loop and see if anything seems off?

<img width="1444" alt="Screenshot 2024-08-08 at 09 47 22" src="https://github.com/user-attachments/assets/1227a668-0d0c-422a-99d1-ae18b7b7027f">

I think it could be due to the inversion that is happening on `expectedSyncing = !value`. I'm not quite sure to understand this bit, could you clarify?